enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/helper/ErrorHandler.java

```diff
@@ -62,4 +83,9 @@ private void throwIfException()
             throw runtimeException;
         }
     }
+
+    public interface ThrowingRunnable
```

Comment:
This would be the fourth implementation of `ThrowingRunnable`. Time to move them into `org.neo4j.function`?

Reply:
Or make ThrowingAction extend Throwable maybe? I would like to let it slide this time tough if you are OK with doing that in a separate PR? Considering the timing.